src/ua/gorshkov/hw10/Main.java

```diff
@@ -0,0 +1,119 @@
+        Triangle triangle;
+        while(true){
+            try {
+                System.out.print("Circle radius = ");
+                circleRadius = scanner.nextDouble();
+                System.out.print("Square side length = ");
+                squareSideLength = scanner.nextDouble();
+                System.out.print("Triangle first side length = ");
+                triangleFirstSideLength = scanner.nextDouble();
+                System.out.print("Triangle second side length = ");
+                triangleSecondSideLength = scanner.nextDouble();
+                System.out.print("Triangle angle value (in radians) = ");
+                triangleAngleValueInRadians = scanner.nextDouble();
+
+                circle = new Circle(circleRadius);
+                square = new Square(squareSideLength);
+                triangle = new Triangle(triangleFirstSideLength,
+                        triangleSecondSideLength,
+                        triangleAngleValueInRadians);
+                break;
+            } catch (InvalidValueException e) {
+                System.out.println(e.getMessage());
+                System.out.println("Enter the values again.");
+            }
+        }
+
+        System.out.println("Circle perimeter = " + circle.perimeter() + ", Circle area = " + circle.area());
+        System.out.println("Square perimeter = " + square.perimeter() + ", Square area = " + square.area());
+        System.out.println("Triangle perimeter = " + triangle.perimeter() + ", Triangle area = " + triangle.area());
+
+        System.out.println();
+        System.out.println("Calculating area by Heron's formula.");
+        double triangleFirstSideLengthInHeronsFormula;
+        double triangleSecondSideLengthInHeronsFormula;
+        double triangleThirdSideLengthInHeronsFormula;
+
+        double areaOfTriangleByHeronsFormula;
+        while(true){
+            try {
+                System.out.print("Triangle first side length = ");
+                triangleFirstSideLengthInHeronsFormula = scanner.nextDouble();
+                System.out.print("Triangle second side length = ");
+                triangleSecondSideLengthInHeronsFormula = scanner.nextDouble();
+                System.out.print("Triangle third side length = ");
+                triangleThirdSideLengthInHeronsFormula = scanner.nextDouble();
+
+                areaOfTriangleByHeronsFormula =
+                        triangle.area(triangleFirstSideLengthInHeronsFormula,
+                                triangleSecondSideLengthInHeronsFormula,
+                                triangleThirdSideLengthInHeronsFormula);
+                System.out.println("Area of triangle by Herons formula = " + areaOfTriangleByHeronsFormula);
+                break;
+            } catch (InvalidValueException | ImpossibleTriangleException e) {
+                System.out.println(e.getMessage());
+                System.out.println("Enter the values again.");
+            }
+        }
```

Comment:
Main method is a pretty big, please, try to move this logic to the single methods